README.md

```diff
@@ -28,14 +28,222 @@ In addition, that format would include optional **signing** of the resources, wh
+   from the spec as it's complex with limited use cases.
+
+Note that this is just an explainer, **not a specification**. We'll add more
+precision when we translate it to a spec.
+
+### Overall format
+
+The package is structured as follows. All "offsets" are 8-byte little-endian
+integers representing a byte offset from the start of the initial magic number.
+
+1. A magic number: `F0 9F 8C 90 F0 9F 93 A6` (🌐📦 in UTF-8).
+2. The length of the package (all bytes from the first of the initial magic
+   number to the last of the final magic number) as a little-endian 8-byte
+   integer.
+3. A 1-byte count of metadata sections followed by that many of the following
+   entries. Repeated entry IDs and non-monotonically-increasing offsets cause
+   the package to fail to parse.
+   * The byte `01` indicating an [index](#index), followed by the offset of the
+     start of the index.
+   * The byte `02` indicating a block of [certificates](#certificates), followed
+     by the offset of the start of the certificates.
+   * The byte `03` indicating a [manifest](#manifest), followed by the offset of
+     the start of the manifest.
+
+   Unknown entry IDs are assumed to be followed by an offset, and the pair must
+   be ignored.
+4. The offset of the [main content](#main-content) of the package.
+5. The metadata sections promised above.
+6. A sequence of HTTP responses constituting the main content of the package.
+   Mismatches with the manifest usually cause the package to fail to parse, as
+   described in the [manifest](#manifest) section, but mismatches with the index
+   might not.
+7. The length of the package, repeated from earlier. This allows tools to
+   operate on the package even if it's been prepended with a self-extracting
+   executable.
+8. The same magic number: `F0 9F 8C 90 F0 9F 93 A6` (🌐📦 in UTF-8).
+
+The MIME type of this format is `application/package`. TODO: Is that right?
+
+Note that this top-level information is *not signed*, and so can't be trusted.
+Only information in the manifest and below can be trusted.
+
+The manifest and certificates are used to sign the contents of the package. The
+index is used when the package is stored on disk, to find resources quickly. And
+the start page should be loaded when a UA is navigated to the package itself.
+
+### Index
+
+The index is optional, and can be added by a client after the package is
+downloaded.
+
+It contains a series of lines (terminated by `\r\n`) of the form
 
-We also propose to remove the [fragment-based URL schema](https://w3ctag.github.io/packaging-on-the-web/#fragment-identifiers) from the spec as it is not clear what would be the use case supporting it.
+```
+URL offset length
+```
+
+Where offset and length are 64-bit integers, and the URLs are absolute. Offsets
+are interpreted relative to the first byte of the package. All resources in the
+package should be included in the index, even if they come from a different
+origin.
```

Comment:
This needs to be clarified wrt subpackages. If a subpackage has its own index, it should not be duplicated. Realistically, the process of serving resources from the packages should include memory cache of indices of subpackages, so there is no need to flatten the index. If I understand the sentence correctly, of course.